flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -259,6 +255,62 @@ static fir::ExtendedValue arraySectionElementToExtendedValue(
   return fir::factory::componentToExtendedValue(builder, loc, element);
 }
 
+/// Convert the array_load, `load`, to an extended value. If `path` is not
+/// empty, then traverse through the components designated. The base value is
+/// `newBase`. This does not accept an array_load with a slice operand.
+static fir::ExtendedValue arrayLoadExtValue(fir::FirOpBuilder &builder,
```

Comment:
I am not entirely following what is happening in this function. Is this making an ExtendedValue for the array_load itself, or is it actually making an ExtendedValue for an element of the array_load obtained via and array_fetch or array_update ?

It looks like to me this is similar to `arraySectionElementToExtendedValue`, except the base is an array load instead of an ExtendedValue and it is passing a path, which  `arraySectionElementToExtendedValue` will probably also need after F95.

Reply:
This entire area needs to be redone, in my opinion. We have a collection of functions that almost do what is needed, but none of them is complete nor entirely suitable. This function is like a pontoon bridge ... not stable but needed at the moment.